File: calorie-guide-android-client/calorieguide-core/src/main/java/com/example/core/model/Gender.kt
package com.example.core.model

enum class Gender(val value: String) {
    MALE("male"),
    FEMALE("female"),
    OTHER("other");

    companion object {
        fun from(value: String?): Gender? {
            return when(value) {
                MALE.value -> MALE
                FEMALE.value -> FEMALE
                OTHER.value -> OTHER
                else -> null
            }
        }
    }
}

File: calorie-guide-server/src/model/Gender.js
'use strict'

/**
 * Enum object for gender.
 *
 * @type {Readonly<{MALE: string, FEMALE: string, OTHER: string}>}
 * @enum {string}
 */
const Gender = Object.freeze({
    MALE: "male",
    FEMALE: "female",
    OTHER: "other"
})

module.exports = Gender
